compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -792,15 +799,55 @@ execMigrate projectOpts opts0 inFile1_ inFile2_ mbDir = do
         errorOnLeft
             "Cannot decode daml-lf archive"
             (Archive.decodeArchive dalf)
-    getEntry fp dar =
-        maybe (fail $ "Package does not contain " <> fp) pure $
-        findEntryByPath fp dar
     getModule modName pkg =
         maybe
             (fail $ T.unpack $ "Can't find module" <> LF.moduleNameString modName)
             pure $
         NM.lookup modName $ LF.packageModules pkg
 
+-- | Get an entry from a dar or fail.
+getEntry :: FilePath -> Archive -> IO Entry
+getEntry fp dar =
+    maybe (fail $ "Package does not contain " <> fp) pure $
+    findEntryByPath fp dar
+
+-- | Merge two dars. The idea is that the second dar is a delta. Hence, we take the main in the
+-- manifest from the first.
+execMergeDars :: FilePath -> FilePath -> Maybe FilePath -> IO ()
+execMergeDars darFp1 darFp2 mbOutFp = do
+    let outFp = fromMaybe darFp1 mbOutFp
+    bytes1 <- B.readFile darFp1
+    bytes2 <- B.readFile darFp2
+    let dar1 = toArchive $ BSL.fromStrict bytes1
+    let dar2 = toArchive $ BSL.fromStrict bytes2
+    mf <- mergeManifests dar1 dar2
+    let merged =
+            Archive
+                (nubSortOn eRelativePath $ mf : zEntries dar1 ++ zEntries dar2)
+                Nothing
+                BSL.empty
+    BSL.writeFile outFp $ fromArchive merged
+  where
+    mergeManifests dar1 dar2 = do
+        let mfPath = "META-INF/MANIFEST.MF"
+        let dalfNames =
+                nubSort
+                    [ takeFileName p
```

Comment:
Is there a reason why we deduplicate only based on the filename instead of the full relative path? That seems somewhat weird. I guess we never expect dalfs in subdirectories but then we should probably check for that. I would also only deduplicate once and then only filter the deduplicated list here instead of deduplicating once based on the full path and once on the filename.

Reply:
because only the dalf name will end up in the manifest. we will make sure that the dalf names of two packages are always different if they have different content in the next piece of work.